src/box/gc.h

```diff
@@ -347,18 +347,19 @@ void
 gc_trigger_checkpoint(void);
 
 /**
- * Get a reference to @checkpoint and store it in @ref.
+ * Allocate and get a reference to @checkpoint.
  * This will block the garbage collector from deleting
  * the checkpoint files until the reference is released
- * with gc_put_checkpoint_ref().
+ * with gc_unref_checkpoint().
  *
  * @format... specifies a human-readable name that will be
  * used for listing the reference in box.info.gc().
+ *
+ * The function never fails - returned value is always not NULL.
  */
-CFORMAT(printf, 3, 4)
-void
-gc_ref_checkpoint(struct gc_checkpoint *checkpoint,
-		  struct gc_checkpoint_ref *ref, const char *format, ...);
+CFORMAT(printf, 2, 3)
```

Comment:
Nit: `NODISCARD` in order to avoid memory leak

Reply:
Actually, AFAICS, we use it only in vinyl. And I use it for `box_connect_replica` because it returns a class using RAII idiom so discarding it will break many things. What's about returning a pointer, we have a lot of such functions without this attribute, so this one will just look "peculiar".